olp-cpp-sdk-dataservice-read/src/repositories/PartitionsRepository.cpp

```diff
@@ -392,6 +392,14 @@ model::Partitions PartitionsRepository::GetTileFromCache(
   return {};
 }
 
+bool PartitionsRepository::RemoveFromCache(
+    const client::HRN& catalog, const std::string& layer,
+    const std::string& partition_id,
+    const client::OlpClientSettings& settings) {
+  repository::PartitionsCacheRepository repository(catalog, settings.cache);
+  return repository.ClearPartition(partition_id, layer);
```

Comment:
This is not enough and probably not the correct one. The deletion process should be handled in the following way:
* Get partition metadata.
* Find and delete data handle of the partition.
* Find and delete expiration key and value (if any).
* Delete partition metadata.